docs/references/architecture/adr-119-dog-mempool-gossip.md

```diff
@@ -0,0 +1,367 @@
+
+Efforts to port the CAT mempool to CometBFT were documented in [\#2027]. Experimental results on a small testnet from [\#1472] showed that CAT effectively reduces bandwidth usage. However, its impact on latency was not evaluated in those tests. Porting CAT was finally deprioritised in favor of DOG.
+
+### Limiting the number of peers a transaction is forwarded to
+
+CometBFT allows operators to configure `p2p.experimental_max_gossip_connections_to_non_persistent_peers` and `p2p.experimental_max_gossip_connections_to_persistent_peers`  as a maximum number of peers to send  transactions to ([#1558](https://github.com/cometbft/cometbft/pull/1558) and [#1584](https://github.com/cometbft/cometbft/pull/1584)). 
+This reduces bandwitdth compared to when they are disabled but there is no rule to determine which peers transactions
+are not forwarded to. DOG can be looked at as an enahanced, more informed version of this protocol.
+
+A node can use either this or DOG, but not both at the same time.
+
+
+### libp2p's gossipSub protocol
+
+ [gossipsub](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md), the gossip protocol of `libp2p`, aims to reduce the bandwidth used for gossiping by limiting the number of peers a node exchange data with. The peering is controlled by a router keeping track of peering state and information about the topics peers are subscribed to. 
+
+ Peers form meshes, where nodes within a mesh are subscribed to the same topic and the size of the mesh is defined with a configuration parameter. 
+
+ `gossipSub` allows peers to indicate they already have a particular transaction so that other peers do not forward it to them, by sending a `IHAVE` message. That, however this is done for each individual message. DOG on the other hand, instructs peers to stop forwarding anything from a given peer with one message. 
+
+ There is [ongoing work](https://github.com/libp2p/specs/pull/413/files) within `gossipSub` that might achieve something similar. A peer can send a message instructing another peer not to forward anything to it. DOG on the other hand blocks only one route, allowing traffic to continue through the peer.
+
+## Decision
+
+The CometBFT team has decided to implement the protocol on top of the existing mempool
```

Comment:
Is there a plan to make it default eventually?

Reply:
Yes, it will be enabled by default. It only improves the performance; we haven't seen any disadvantage in having it enabled.

Reply:
Should we say so here @hvanz

Reply:
Yes, probably it's better to make it explicit in the ADR.